cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/ShuntConversion.java

```diff
@@ -43,31 +46,48 @@ public void convert() {
         int sections = getSections(p, normalSections);
         sections = Math.abs(sections);
         maximumSections = Math.max(maximumSections, sections);
-        double bPerSection = 0;
-        if (p.containsKey(CgmesNames.B_PER_SECTION)) {
-            bPerSection = p.asDouble(CgmesNames.B_PER_SECTION, 0.0);
-        } else {
+        ShuntCompensatorAdder adder = voltageLevel().newShuntCompensator().setCurrentSectionCount(sections);
+        String shuntType = p.getId("type");
+        if ("LinearShuntCompensator".equals(shuntType)) {
+            double bPerSection = p.asDouble(CgmesNames.B_PER_SECTION, Float.MIN_VALUE);
+            if (bPerSection == 0) {
+                double bPerSectionFixed = Double.MIN_VALUE;
+                fixed(CgmesNames.B_PER_SECTION, "Can not be zero", bPerSection, bPerSectionFixed);
+                bPerSection = bPerSectionFixed;
+            }
+            double gPerSection = p.asDouble("gPerSection", Double.NaN);
+            adder.newLinearModel()
+                    .setBPerSection(bPerSection)
+                    .setGPerSection(gPerSection)
+                    .setMaximumSectionCount(maximumSections)
+                    .add();
+        } else if ("NonlinearShuntCompensator".equals(shuntType)) {
+            ShuntCompensatorNonLinearModelAdder modelAdder = adder.newNonLinearModel();
             PropertyBags ss = context.cgmes().nonlinearShuntCompensatorPoints(id);
-            final int nlsections = sections;
-            double sumSections = ss.stream()
-                    .filter(s -> s.asInt("sectionNumber") <= nlsections)
-                    .map(s -> s.asDouble("b"))
-                    .reduce(0.0, Double::sum);
-            // Convert to a shunt compensator with a single section
-            maximumSections = 1;
-            sections = 1;
-            bPerSection = sumSections;
-        }
-        if (bPerSection == 0) {
-            float bPerSectionFixed = Float.MIN_VALUE;
-            fixed(CgmesNames.B_PER_SECTION, "Can not be zero", bPerSection, bPerSectionFixed);
-            bPerSection = bPerSectionFixed;
+            for (PropertyBag sp : ss) {
+                int sectionNumber = sp.asInt(SECTION_NUMBER);
+                double b = ss.stream().filter(s -> s.asInt(SECTION_NUMBER) <= sectionNumber).map(s -> s.asDouble("b")).reduce(0.0, Double::sum);
+                if (sectionNumber == 0) {
+                    if (b != 0) {
+                        fixed("Shunt.section0", "Susceptance of section 0 should always be zero (disconnected state)", b, 0.);
+                    }
+                    continue;
+                }
+                if (b == 0) {
+                    double bFixed = Double.MIN_VALUE;
```

Comment:
It is wierd that CGMES requires a b equal to zero and that we don't allow it in IIDM. No?

Reply:
This check might actually not be useful for non linear shunt (see above). Maybe there are shunt that can be considered as disconnected in several positions.

Reply:
The check has to be removed (see above).